backend/src/v2/component/launcher_v2.go

```diff
@@ -460,6 +496,9 @@ const OutputMetadataFilepath = "/tmp/kfp_outputs/output_metadata.json"
 // We overwrite this as a DI mechanism for testing getLogWriter.
 var osCreateFunc = os.Create
 
+// We overwrite this as a DI mechanism for testing collectOutputParameters
+var osReadFileFunc = os.ReadFile
```

Comment:
I think it's more customary to use io.Reader and other built in Go interfaces.

Reply:
https://github.com/kubeflow/pipelines/blame/master/backend/src/v2/component/launcher_v2.go#L461 was a shortcut because I didn't want to have to refactor all the nested function signatures in an already complicated PR, but since you're doing that anyway, @jpthompson23, I think it makes sense to handle this DI idiomatically.